### PR TITLE
IRIS-4943 | Fix tests for following

### PR DESCRIPTION
**Pull request by tortila**
[Ticket](https://wikia-inc.atlassian.net/browse/IRIS-4943)

These changes include:
- grouping test cases for following and unfollowing together (reducing total number of testcases)
- making test cases independent of setup or env - creating posts and interacting with them, instead of with first post on the list (which is important when more posts are being created and used at the same time by other tests)
- much needed restructuring of Discussions related Page Objects: removing `AvailablePage`, removing `PageWithPosts` interface in favor of abstract class

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/ReportedPostsAndRepliesPage.java [hunk 1]
```diff
@@ -2,11 +2,12 @@
 
 import com.wikia.webdriver.elements.mercury.components.discussions.common.ErrorMessages;
 import com.wikia.webdriver.elements.mercury.components.discussions.common.Post;
+import com.wikia.webdriver.elements.mercury.components.discussions.common.SignInToFollowModalDialog;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/ReportedPostsAndRepliesPage.java]: MINOR Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/FollowPage.java [hunk 2]
```diff
@@ -2,10 +2,11 @@
 
 import com.wikia.webdriver.elements.mercury.components.discussions.common.NoFollowedPostsMessage;
 import com.wikia.webdriver.elements.mercury.components.discussions.common.Post;
+import com.wikia.webdriver.elements.mercury.components.discussions.common.SignInToFollowModalDialog;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/FollowPage.java]: MINOR Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [Post.java#L7](https://github.com/Wikia/selenium-tests/blob/cfb845116a914e89af873e9dd14208f2065405d1/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java#L7): Remove this unused import 'lombok.SneakyThrows'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [PostDetailsPage.java#L6](https://github.com/Wikia/selenium-tests/blob/cfb845116a914e89af873e9dd14208f2065405d1/src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PostDetailsPage.java#L6): Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 4 issues
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [Post.java#L7](https://github.com/Wikia/selenium-tests/blob/ff8b9d29e6549853376f2fb8b765ce6322ecb324/src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/Post.java#L7): Remove this unused import 'lombok.SneakyThrows'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [PostDetailsPage.java#L6](https://github.com/Wikia/selenium-tests/blob/ff8b9d29e6549853376f2fb8b765ce6322ecb324/src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PostDetailsPage.java#L6): Remove this unused import 'com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject'. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUselessImportCheck)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java [hunk 3]
```diff
@@ -1,8 +1,30 @@
...
+import com.wikia.webdriver.elements.mercury.components.discussions.common.PostEntity;
+import com.wikia.webdriver.elements.mercury.components.discussions.common.SignInToFollowModalDialog;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+
+import java.util.Optional;
+
+public abstract class PageWithPosts extends WikiBasePageObject {
+
+  public Optional<PostEntity> getPostById(String postId) {
+    return Optional.ofNullable(
```
Comment (TK-999) [inline src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java]: This outer optional shouldnt be needed here

**Pull request merged**